dallinger/frontend/static/scripts/dallinger.js

```diff
@@ -237,16 +237,14 @@ var create_participant = function() {
 var lock = false;
 
 var submitResponses = function () {
-    submitNextResponse(0);
-    submitAssignment();
+    submitNextResponse(0, submitAssignment);
 };
 
 var submit_responses = function () {
     submitResponses();
-    submitAssignment();
 };
 
-var submitNextResponse = function (n) {
+var submitNextResponse = function (n, callback = function(){}) {
 
     // Get all the ids.
     var ids = $("form .question select, input, textarea").map(
```

Comment:
This selector looks bad to me. It should probably be `$("form .question").find("select, input, textarea")` or some variant so that you are only looking at inputs and textareas in a `.question` wrapper.